pkg/server/server.go

```diff
@@ -393,6 +393,9 @@ func (s *Server) VerifyEmail(c *gin.Context, redirectURL string) {
 		tmp := jwtUserCredentials.GetUnverifiedEmail()
 		jwtUserCredentials.Email = &tmp
 		jwtUserCredentials.UnverifiedEmail = nil
+		if jwtUserCredentials.GetKind() == models.LocalAuth {
+			jwtUserCredentials.UserName = &tmp
+		}
```

Comment:
We shouldn't assign the address this way. Better way is to initialise the username field and assign the value.
*username = tmp